extensions/wikia/WallNotifications/WallNotificationEntity.class.php

```diff
@@ -1,61 +1,116 @@
 <?php
 
-class WallNotificationEntity {
+use Wikia\Logger\WikiaLogger;
 
+class WallNotificationEntity {
+	/** @var int */
 	public $id;
-	/* data stored in memcache */
-	public $data;
-	public $data_non_cached;
-	/* this data is here only after you create this object
-	 * when recreating object from memcache this will be empty
-	 */
-	public $data_noncached;
+
+	/** @var stdClass data stored in memcache */
+	public $data = null;
+
+	public $parentTitleDbKey = '';
+	public $msgText = '';
+	public $threadTitleFull = '';
 
 	/*
 	 *	Public Interface
 	 */
 
 	/**
-	 * @param Revision $rev
-	 * @param $wikiId
-	 * @param bool $master
+	 * Create a new object from an existing revision object.
+	 *
+	 * @param Revision $rev A revision object
+	 * @param bool $useMasterDB Whether to query the MASTER DB on Title lookup.
 	 *
 	 * @return WallNotificationEntity
 	 */
```

Comment:
If you have a ```Revision``` object already you don't need to specify a city ID.  You should be able to get the city from ```$wgCityId```.